test/functional/test_framework/blocktools.py

```diff
@@ -44,14 +44,15 @@
 # From BIP141
 WITNESS_COMMITMENT_HEADER = b"\xaa\x21\xa9\xed"
 
-def create_block(hashprev, coinbase, ntime=None):
+def create_block(hashprev, coinbase, block_height, ntime=None):
```

Comment:
Modifying the function signature like this is begging for silent bugs.

Instead, add `block_height` to the end, and always specify it by name (enforce with PEP 3102; ie, `(hashprev, coinbase, ntime=None, *, block_height)`).

Reply:
Ah that's what the random asterixes are for! :) Thanks

Reply:
Another option is to make ntime non optional.